test/scripts/deploy.test.ts

```diff
@@ -0,0 +1,122 @@
+import chai, { expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import { Contract, ContractFactory } from 'ethers';
+import { ethers } from 'hardhat';
+import sinon from 'sinon';
+import fs from 'fs';
+import * as hardhat from 'hardhat';
+import {
+  deployContracts,
+  generateJsonConfig,
+} from '../../scripts/modules/deploy';
+
+chai.use(chaiAsPromised);
+
+describe('Deploy Script', function () {
+  describe('deployContracts', function () {
+    const testAddress = '0x145845fd06c85B7EA1AA2d030E1a747B3d8d15D7';
+    beforeEach(function () {
+      const contract = new Contract(testAddress, []);
+      const contractFactoryStub = sinon.createStubInstance(ContractFactory);
+      sinon.stub(ethers, 'getContractFactory').resolves(contractFactoryStub);
+      contractFactoryStub.deploy.resolves(contract);
+    });
+
+    afterEach(function () {
+      sinon.restore();
+    });
+
+    it('should deploy all contracts', async function () {
+      const result = await deployContracts();
+      expect(result).to.have.all.keys(
+        'Penalizer',
+        'RelayHub',
+        'SmartWallet',
+        'SmartWalletFactory',
+        'SmartWalletDeployVerifier',
+        'SmartWalletRelayVerifier',
+        'CustomSmartWallet',
+        'CustomSmartWalletFactory',
+        'CustomSmartWalletDeployVerifier',
+        'CustomSmartWalletRelayVerifier',
+        'UtilToken'
+      );
+    });
+
+    it('should deploy contracts with valid addresses', async function () {
+      const result = await deployContracts();
+      Object.values(result).forEach((value) => {
+        expect(value).to.eq(testAddress);
```

Comment:
```suggestion
        expect(value, value).to.eq(testAddress);
```
to show which one failed if any